### DISCUSSION TITLE
Setup questions regarding login and 2FA

### DISCUSSION THREAD

**Discussion by popy2k14** [alandtse/alexa_media_player#1535] [Q&A]
hi guys.

I am struggling with the custom integration alexa media player, yesterday it did not renew the login/cookie and all alexas stopped working from within HA. The integration was RED in the HA gui. After relogin with my credentials (and dropped the !secret and also entrys in secrets.yaml), it started working again.
Please don't blame me, i have readed the wiki, watched YT videos but want to make it right, so next time i dont have to manually do something.

Before HA i was on fhem and used their similar integration without an issue. I think they are using the same unofficial alexa api. I had not activated 2FA and it worked.

My initial setup of the HA integration was (which failed yesterday to relogin):

- Activated two factor over SMS, dont want an auth app
- Readed the wiki multiple times -> sadly the instructions are not really clear if the "normal" login proxy method without 2FA is working to renew the cookie (also when 2FA with SMS is activated)
- used "!secret amazon_user" & "!secret amazon_password" directly over the gui (wiki states the gui method is the recommended one)
- Used Poxy method to initially login, and entered the 2FA SMS code on the amazon website

This worked until yesterday.

So here are my questions:
- Does the cutsom integration with the login proxy method (and SMS 2FA activated in amazon account) renew the login automatically?
- Do i really have to activate 2FA with an Authenticator app in my amazon account? Or just copy the my key/code like described (the bold value) and enter it along with my credentials?
Asking because dont want to activate login with an auth app.
- How can i check that the integration is able to renew the needed cookie informations to stay logged in.

Thank you

**Comment by GeorgeIoak** [comment]
+1 on this request as I too have been struggling with the proper and least painful way to keep this integration working. I don't know if it's normal but quite regularly I have to reconfigure and that is hit or miss if it works as well as hit or miss if it shows a captcha or not. Sometimes a get texts from Amazon with a OTP which I believe is coming from this integration as they seem to happen when HA lets me know that I need to reconfigure the integration.

Maybe someone who really knows all the options and "best" methods could review the Wiki and see if it's really clear enough.
